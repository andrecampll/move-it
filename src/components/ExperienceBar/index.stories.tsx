import { Story, Meta } from '@storybook/react/types-6-0';
import ExperienceBar from './index';

export default {
  title: 'ExperienceBar',
  component: ExperienceBar,
} as Meta;

export const Basic: Story = () => <ExperienceBar />;
