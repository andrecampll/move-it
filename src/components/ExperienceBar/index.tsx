import { Wrapper } from './styles'

const ExperienceBar = () => (
  <Wrapper data-testid="experience-bar">
    <header>
      <span>0 xp</span>
      <div>
        <div style={{ width: '50%' }} />

        <span style={{ left: '50%' }}>300 xp</span>
      </div>
      <span>600 xp</span>
    </header>
  </Wrapper>
)

export default ExperienceBar
