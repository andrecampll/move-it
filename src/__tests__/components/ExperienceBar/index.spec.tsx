import { render, screen } from '@testing-library/react'
import ExperienceBar from '../../../components/ExperienceBar'

describe('ExperienceBar', () => {
  it('should be able to render', () => {
    const { container } = render(<ExperienceBar />)

    expect(screen.getByTestId('experience-bar')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
