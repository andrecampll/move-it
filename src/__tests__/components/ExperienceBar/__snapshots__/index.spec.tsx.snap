// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceBar should be able to render 1`] = `
.c0 header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 header span {
  font-size: 1rem;
}

.c0 header > div {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 4px;
  border-radius: 4px;
  background: var(--gray-line);
  margin: 0 1.5rem;
  position: relative;
}

.c0 header > div > div {
  height: 4px;
  border-radius: 4px;
  background: var(--green);
}

.c0 header > div > span {
  position: absolute;
  top: 12px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

<div
  class="c0"
  data-testid="experience-bar"
>
  <header>
    <span>
      0 xp
    </span>
    <div>
      <div
        style="width: 50%;"
      />
      <span
        style="left: 50%;"
      >
        300 xp
      </span>
    </div>
    <span>
      600 xp
    </span>
  </header>
</div>
`;
